feature_engine/timeseries/forecasting/lag_features.py

```diff
@@ -243,3 +289,30 @@ def _get_new_features_name(self) -> List:
             ]
 
         return feature_names
+
+    def _agg_freq_lags(
+        self,
+        grouped_df: pd.core.groupby.generic.DataFrameGroupBy,
+        freq: Union[str, List[str]],
+    ) -> Union[pd.Series, pd.DataFrame]:
+        """_summary_
+
+        Parameters
+        ----------
+        grouped_df : pd.core.groupby.generic.DataFrameGroupBy
+            dataframe of groups
+        freq : Union[str, List[str]]
+            Offset to use from the tseries module or time rule.
+
+        Returns
+        -------
+        Union[pd.Series, pd.DataFrame]
+            lag feature or dataframe of lag features
+        """
+        tmp_data = []
+        for _, group in grouped_df:
```

Comment:
Do we need to loop over the groups to apply the lags? pandas does the lags per group automatically.

Reply:
I tried many approaches to simplify this approach, but it is only working when using `periods` argument with `shift()` method like the line in 231
, however when using `freq` argument with `shift()` method it doesn't work, so I used loop to make it work.
kindly advice if we can simplify it.